int32 speed
int32 direction
